src/agent/services/lazer_exporter.rs

```diff
@@ -0,0 +1,227 @@
+    let client = Client::new();
+    let response = client
+        .get(url)
+        .send()
+        .await?
+        .text()
+        .await?;
+    Ok(serde_json::from_str(&response)?)
+}
+
+#[instrument(skip(config, state))]
+pub fn lazer_exporter(config: Config, state: Arc<state::State>) -> Vec<JoinHandle<()>>
+{
+    // TODO: add loop to handle relayer failure/retry
+    let mut handles = Vec::new();
+    handles.push(tokio::spawn(lazer_exporter::lazer_exporter(config.clone(), state)));
+    handles
+}
+
+mod lazer_exporter {
+    use std::collections::HashMap;
+    use std::num::NonZeroI64;
+    use std::sync::Arc;
+    use std::time::Duration;
+    use futures_util::StreamExt;
+    use pyth_lazer_protocol::publisher::PriceFeedDataV1;
+    use pyth_lazer_protocol::router::{Price, PriceFeedId, TimestampUs};
+    use tokio_stream::StreamMap;
+    use crate::agent::services::lazer_exporter::{Config, connect_to_relayers, fetch_symbols, SymbolResponse};
+    use crate::agent::state::local::LocalStore;
+
+    pub async fn lazer_exporter<S>(config: Config, state: Arc<S>)
+    where
+        S: LocalStore,
+        S: Send + Sync + 'static,
+    {
+        let mut failure_count = 0;
+        let retry_duration = Duration::from_secs(1);
+
+        loop {
+            run(&config, state.clone()).await;
+
+            failure_count += 1;
+            tracing::error!("Lazer exporter failed {} times; retrying in {:?}", failure_count, retry_duration);
+            tokio::time::sleep(retry_duration).await;
+
+            // TODO: Back off or crash altogether on persistent failure
+        }
+    }
+
+    async fn run<S>(config: &Config, state: Arc<S>)
+    where
+        S: LocalStore,
+        S: Send + Sync + 'static,
+    {
+        // TODO: Re-fetch on an interval?
+        let lazer_symbols: HashMap<String, SymbolResponse> = match fetch_symbols(&config.history_url).await {
+            Ok(symbols) => symbols.into_iter().filter_map(|symbol| {
+                symbol.hermes_id.clone().map(|id| (id, symbol))
+            }).collect(),
+            Err(e) => {
+                tracing::error!("Failed to fetch Lazer symbols: {e:?}");
+                return;
+            }
+        };
+
+        // Establish relayer connections
+        // Relayer will drop the connection if no data received in 5s
+        let (mut relayer_sender, relayer_receivers) = connect_to_relayers(&config)
+            .await.expect("failed to connect to relayers");
+        let mut stream_map = StreamMap::new();
+        for (i, receiver) in relayer_receivers.into_iter().enumerate() {
+            stream_map.insert(config.relayer_urls[i].clone(), receiver);
+        }
+
+        let mut publish_interval = tokio::time::interval(config.publish_interval_duration);
+
+        loop {
+            tokio::select! {
+                _ = publish_interval.tick() => {
+                    for (identifier, price_info) in state.get_all_price_infos().await {
```

Comment:
since we want to run this in high frequency i want to make sure this doesn't become expensive as it might copy the data.